core/templates/components/common-layout-directives/navigation-bars/top-navigation-bar.component.ts

```diff
@@ -0,0 +1,368 @@
+       this.isSuperAdmin = userInfo.isSuperAdmin();
+       this.userIsLoggedIn = userInfo.isLoggedIn();
+       this.username = userInfo.getUsername();
+       if (this.username) {
+         this.profilePageUrl = this.urlInterpolationService.interpolateUrl(
+           '/profile/<username>', {
+             username: this.username
+           });
+       }
+
+       if (this.userIsLoggedIn) {
+         this.navigationBackendApiService.showUnseenNotifications();
+       }
+     });
+
+     for (var i = 0; i < this.NAV_ELEMENTS_ORDER.length; i++) {
+       this.navElementsVisibilityStatus[this.NAV_ELEMENTS_ORDER[i]] = true;
+     }
+
+     this.directiveSubscriptions.add(
+       this.windowDimensionsService.getResizeEvent().subscribe(evt => {
+         this.windowIsNarrow = this.windowDimensionsService.isWindowNarrow();
+         // If window is resized larger, try displaying the hidden
+         // elements.
+         if (
+           this.currentWindowWidth < this.windowDimensionsService.getWidth()) {
+           for (var i = 0; i < this.NAV_ELEMENTS_ORDER.length; i++) {
+             if (
+               !this.navElementsVisibilityStatus[
+                 this.NAV_ELEMENTS_ORDER[i]]) {
+               this.navElementsVisibilityStatus[this.NAV_ELEMENTS_ORDER[i]] =
+                 true;
+             }
+           }
+         }
+
+         // Close the sidebar, if necessary.
+         this.sidebarStatusService.closeSidebar();
+         this.sidebarIsShown = this.sidebarStatusService.isSidebarShown();
+         this.currentWindowWidth = this.windowDimensionsService.getWidth();
+         this.debouncerService.debounce(this.truncateNavbar, 500);
+       })
+     );
+     // The function needs to be run after i18n. A timeout of 0 appears
+     // to run after i18n in Chrome, but not other browsers. The
+     // will check if i18n is complete and set a new timeout if it is
+     // not. Since a timeout of 0 works for at least one browser,
+     // it is used here.
+     setTimeout(this.truncateNavbar, 0);
+   }
+
+   async getProfileImageDataAsync(): Promise<void> {
+     let dataUrl = await this.userService.getProfileImageDataUrlAsync();
+     this.profilePictureDataUrl = decodeURIComponent(dataUrl);
+   }
+
+
+   getStaticImageUrl(imagePath: string): string {
+     return this.urlInterpolationService.getStaticImageUrl(imagePath);
+   }
+
+   onLoginButtonClicked(): void {
+     this.userService.getLoginUrlAsync().then(
+       (loginUrl) => {
+         if (loginUrl) {
+           this.siteAnalyticsService.registerStartLoginEvent('loginButton');
+           setTimeout(() => {
+             this.windowRef.nativeWindow.location.href = loginUrl;
+           }, 150);
+         } else {
+           this.windowRef.nativeWindow.location.reload();
+         }
+       }
+     );
+   }
+   onLogoutButtonClicked(): void {
+     this.windowRef.nativeWindow.localStorage.removeItem(
+       'last_uploaded_audio_lang');
+   }
+   /**
```

Comment:
can you ensure that there exist a single empty line, between every two consecutive functions?

Reply:
done